### ISSUE TITLE
Different numeric datatype of literal depending on return type

### ISSUE THREAD

**Issue by LorenzBuehmann** [openlink/virtuoso-opensource#454]
Tested against officila DBpedia endpoint (http://dbpedia.org/sparql)

When retrieving triples via CONSTRUCT queries, and among those triples there are literal values that are supposed to be xsd:double, depending on the chosen return type (TURTLE vs. N-TRIPLES) the literals have a different return type.

A [test query](http://dbpedia.org/sparql?default-graph-uri=http%3A%2F%2Fdbpedia.org&qtxt=construct+%7B+%3Fs+%3Chttp%3A%2F%2Fdbpedia.org%2Fontology%2Fheight%3E+%3Fo+.+%7D+%7B+%3Fs+%3Chttp%3A%2F%2Fdbpedia.org%2Fontology%2Fheight%3E+%3Fo+.+%7D+limit+1&format=text%2Fplain&timeout=30000&debug=on) is

```
CONSTRUCT { ?s <http://dbpedia.org/ontology/height> ?o . } { ?s <http://dbpedia.org/ontology/height> ?o . } LIMIT 1
```

TURTLE result:

```
@prefix xsd:    <http://www.w3.org/2001/XMLSchema#> .
@prefix dbo:    <http://dbpedia.org/ontology/> .
@prefix dbr:    <http://dbpedia.org/resource/> .
dbr:Adam_Yahiye_Gadahn  dbo:height  1.5240000000000000213 .
```

N-TRIPLE result:

```
<http://dbpedia.org/resource/Adam_Yahiye_Gadahn>    <http://dbpedia.org/ontology/height>    "1.524"^^<http://www.w3.org/2001/XMLSchema#double> .
```

Literals like 1.3 are supposed to be of type xsd:decimal in Turtle.

When querying for resources with the given literal, xsd:double has to be used, so I guess there is a bug in the Turtle serializer.

**Comment by HughWilliams** [comment]
We are looking into this reported issue ...

**Comment by jindrichmynarz** [comment]
@LorenzBuehmann: Have you resolved this issue somehow?

It seems that Virtuoso's serialization of `xsd:double` to Turtle is broken. While serializing the following query to N-Triples or RDF/XML preserves the `xsd:double` data type, Turtle export casts it to `xsd:decimal`:

``` sparql
PREFIX :    <http://example.com/>
PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>

CONSTRUCT {
  [] :p "1.23"^^xsd:double.
}
WHERE {}
```

**Comment by LorenzBuehmann** [comment]
@jindrichmynarz No, I just reported this bug, but I don't know whether this has been fixed in any new versions of Virtuoso.
At least the version on which DBpedia is hosted still does return the implicit `xsd:decimal` for the Turtle format.

**Comment by HughWilliams** [comment]
this issue is on the agenda for development to look into ...

**Comment by jindrichmynarz** [comment]
Eventually, I worked around this by requesting data in N-Triples instead of Turtle, since `xsd:double`s are correctly preserved in this serialization.

**Comment by LorenzBuehmann** [comment]
@jindrichmynarz I know that this works for N-Triples (see my initial post). That's why I reported this issue as I've seen that there is a difference between the N-Triples and Turtle serialization. 
I hope that it will be fixed together with the more critical https://github.com/openlink/virtuoso-opensource/issues/569 in the next version.
